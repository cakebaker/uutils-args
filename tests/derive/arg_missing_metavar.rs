use uutils_args::{Arguments, Options};

struct Complicated {
    // Doesn't "impl Default". Oops!
}

#[derive(Arguments)]
enum Arg {
    #[arg("--foo")]
    Something(Complicated),
}

struct Settings {}

impl Options<Arg> for Settings {
    fn apply(&mut self, _arg: Arg) -> Result<(), uutils_args::Error> {
        Ok(())
    }
}

fn main() {
    Settings {}.parse(std::env::args_os()).unwrap();
}
