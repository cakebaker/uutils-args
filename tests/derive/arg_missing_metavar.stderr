error[E0277]: the trait bound `Complicated: Default` is not satisfied
 --> tests/derive/arg_missing_metavar.rs:7:10
  |
7 | #[derive(Arguments)]
  |          ^^^^^^^^^ the trait `Default` is not implemented for `Complicated`
  |
  = note: this error originates in the derive macro `Arguments` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `Complicated` with `#[derive(Default)]`
  |
3 + #[derive(Default)]
4 | struct Complicated {
  |
